src/ch/blinkenlights/android/vanilla/CoverCache.java

```diff
@@ -199,7 +199,12 @@ public String toString() {
 		/**
 		 * Priority-ordered list of possible cover names
 		 */
-		private final static Pattern[] COVER_MATCHES = { Pattern.compile("(?i).+/(COVER|ALBUM)\\.(JPE?G|PNG)$"), Pattern.compile("(?i).+/(CD|FRONT|ARTWORK)\\.(JPE?G|PNG)$"), Pattern.compile("(?i).+\\.(JPE?G|PNG)$") };
+		private final static Pattern[] COVER_MATCHES = {
+			    Pattern.compile("(?i).+/(COVER|ALBUM)\\.(JPE?G|PNG)$"),
+			    Pattern.compile("(?i).+/ALBUMART(SMALL)?\\.(JPE?G|PNG)$"),
+			    Pattern.compile("(?i).+/ALBUMART_\\{[-0-9A-F]+\\}_(SMALL|LARGE)\\.(JPE?G|PNG)$"),
```

Comment:
That looks reasonable, but do we ever want to pick up the 'small' cover version? (I assume WMP writes both?)

Maybe we could just unify them in something like
```
(?i).+/ALBUMART.*(?!SMALL).*\\.(JPE?G|PNG)$
```
to match everything which begins with 'albumart' but doesn't have the word 'small' in it.
(But not sure if this regexp works as expected in java)

Reply:
1. Aren't small versions used in the library view? Does Vanilla scale the images?
1. SMALL is always at the end so the latter `.*` is unnecessary. I don't recommend regular expressions with more than one `.*`, they are less efficient.
1. If you don't want match small versions the expression should be:
```
(?i).+/ALBUMART(_\\{[-0-9A-F]+\\}_LARGE)?\\.(JPE?G|PNG)$
```

Reply:
> Aren't small versions used in the library view? Does Vanilla scale the images?

Yes, we do not depend on pre-scaled images: The scaling is always done (+ cached) internally

> SMALL is always at the end so the latter .* is unnecessary. 

That's true, but i wouldn't mind the regexp to be 'relaxed' to also pick other items which have a similar name. Also: The regexp doesn't get executed that often (only on cache miss of artwork), so i wouldn't worry much about the performance ;-)

But the regexp in #3 also looks fine to me